In [1]:
import pandas as pd 
import numpy as np 
import plotly.graph_objects as go 

In [2]:
df = pd.read_csv(r"C:\Users\RITHIK KUMAR\Downloads\archive\train.csv")
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales
0,1,CA-2017-152156,08/11/2017,11/11/2017,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.0,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600
1,2,CA-2017-152156,08/11/2017,11/11/2017,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.0,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400
2,3,CA-2017-138688,12/06/2017,16/06/2017,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,California,90036.0,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200
3,4,US-2016-108966,11/10/2016,18/10/2016,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.0,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775
4,5,US-2016-108966,11/10/2016,18/10/2016,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.0,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9800 entries, 0 to 9799
Data columns (total 18 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9800 non-null   int64  
 1   Order ID       9800 non-null   object 
 2   Order Date     9800 non-null   object 
 3   Ship Date      9800 non-null   object 
 4   Ship Mode      9800 non-null   object 
 5   Customer ID    9800 non-null   object 
 6   Customer Name  9800 non-null   object 
 7   Segment        9800 non-null   object 
 8   Country        9800 non-null   object 
 9   City           9800 non-null   object 
 10  State          9800 non-null   object 
 11  Postal Code    9789 non-null   float64
 12  Region         9800 non-null   object 
 13  Product ID     9800 non-null   object 
 14  Category       9800 non-null   object 
 15  Sub-Category   9800 non-null   object 
 16  Product Name   9800 non-null   object 
 17  Sales          9800 non-null   float64
dtypes: float

In [7]:
highest_performing_region = df.groupby("Region")["Sales"].sum().reset_index().sort_values(by = "Sales", ascending=False)
highest_performing_region

,Region,Sales
3,West,710219.6845
1,East,669518.7260
0,Central,492646.9132
2,South,389151.4590


In [9]:
which_catogrey_make_more = df.groupby("Category")["Sales"].sum().reset_index().sort_values(by = "Sales", ascending=False)
which_catogrey_make_more

,Category,Sales
2,Technology,827455.8730
0,Furniture,728658.5757
1,Office Supplies,705422.3340


In [12]:
df["Sub-Category"][:10]

0      Bookcases
1         Chairs
2         Labels
3         Tables
4        Storage
5    Furnishings
6            Art
7         Phones
8        Binders
9     Appliances
Name: Sub-Category, dtype: object

In [17]:
which_sub_category_make_more = df.groupby(["Category", "Sub-Category"])["Sales"].sum().reset_index().sort_values(by = "Sales", ascending=False)
which_sub_category_make_more[:5]

,Category,Sub-Category,Sales
16,Technology,Phones,327782.448
1,Furniture,Chairs,322822.731
11,Office Supplies,Storage,219343.392
3,Furniture,Tables,202810.628
6,Office Supplies,Binders,200028.785


In [19]:
which_sub_category_make_more[-5:]

,Category,Sub-Category,Sales
12,Office Supplies,Supplies,46420.308
5,Office Supplies,Art,26705.410
7,Office Supplies,Envelopes,16128.046
9,Office Supplies,Labels,12347.726
8,Office Supplies,Fasteners,3001.960


In [25]:
df["Month"] = pd.to_datetime(df["Order Date"], dayfirst=True).dt.month
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Month
0,1,CA-2017-152156,08/11/2017,11/11/2017,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.0,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,11
1,2,CA-2017-152156,08/11/2017,11/11/2017,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.0,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,11
2,3,CA-2017-138688,12/06/2017,16/06/2017,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,California,90036.0,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,6
3,4,US-2016-108966,11/10/2016,18/10/2016,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.0,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,10
4,5,US-2016-108966,11/10/2016,18/10/2016,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.0,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,10


In [26]:
seasonal_sales_report = df.groupby("Month").agg({
    "Sales" : ["sum", "mean", "median"]
}).reset_index()

seasonal_sales_report.columns = [ "_".join(col) for col in seasonal_sales_report]
seasonal_sales_report

,Month_,Sales_sum,Sales_mean,Sales_median
0,1,94291.6296,257.627403,55.084
1,2,59371.1154,199.902745,59.970
2,3,197573.5872,290.549393,58.650
3,4,136283.0006,207.432269,47.976
4,5,154086.7237,212.533412,49.568
5,6,145837.5233,211.052856,51.980
6,7,145535.6890,208.802997,58.320
7,8,157315.9270,227.007110,54.224
8,9,300103.4117,221.642106,48.940
9,10,199496.2947,246.596162,52.680


In [33]:
fig = go.Figure()

fig.add_trace(
    go.Scatter(
        x = seasonal_sales_report["Month_"],
        y = seasonal_sales_report["Sales_sum"],
        mode = "lines+markers"
    )
)

fig.update_layout(
    xaxis_title = "Months in Numbers",
    yaxis_title = "Sales in each month",
    title = "Line chart of monthly sales"
)
fig.show()

In [35]:
customer_segment_based_revenue = df.groupby("Segment")["Sales"].mean().reset_index().sort_values(by = "Sales", ascending=False)
customer_segment_based_revenue

,Segment,Sales
2,Home Office,243.403309
1,Corporate,233.150720
0,Consumer,225.065777


In [36]:
customer_segment_based_revenue = df.groupby("Segment")["Sales"].sum().reset_index().sort_values(by = "Sales", ascending=False)
customer_segment_based_revenue

,Segment,Sales
0,Consumer,1.148061e+06
1,Corporate,6.884941e+05
2,Home Office,4.249822e+05


In [44]:
loss_selling_products = df.groupby("Product Name").agg({
    "Sales" : ["sum","mean"],
    "Order Date" : ["count"]
}).reset_index()
loss_selling_products.columns = [ "_".join(col) for col in loss_selling_products]
loss_selling_products.sort_values(by = ["Sales_sum","Order Date_count"], ascending=[True,True])[:20]

,Product Name_,Sales_sum,Sales_mean,Order Date_count
604,Eureka Disposable Bags for Sanitaire Vibra Gro...,1.624,1.624,1
208,Avery 5,5.760,5.760,1
1792,Xerox 20,6.480,6.480,1
783,Grip Seal Envelopes,7.072,7.072,1
97,Acme Serrated Blade Letter Opener,7.632,7.632,1
253,Avery Hi-Liter Pen Style Six-Color Fluorescent...,7.700,7.700,1
249,Avery Hi-Liter Comfort Grip Fluorescent Highli...,7.800,3.900,2
1779,Xerox 1989,7.968,7.968,1
27,4009 Highlighters,8.040,8.040,1
1508,Stockwell Gold Paper Clips,8.096,4.048,2


### Revenue Performance by Region

- **Top Performing Region:** West  
  - Highest total revenue (~710K)  
  - Primary driver of business revenue  

- **Underperforming Region:** South  
  - Lowest total revenue (~389K)  
  - Potential area for improvement  

### Insight
West is leading significantly, while South is lagging behind. This gap suggests differences in demand, pricing, or operational efficiency across regions.

### Sales Volume Analysis

#### Highest Sales by Category
- **Technology**  
  - Total Sales: ~827K  
  - Top revenue-driving category

#### Highest Sales by Sub-Category
- **Phones (Technology)**  
  - Total Sales: ~327K  
  - Highest contributing sub-category overall

---

### Insight
Technology is the dominant category in terms of sales volume, with Phones being the primary driver within it. This indicates strong demand for tech products, especially mobile devices.

### Seasonal Sales Trend Analysis

#### Peak Months
- **November (Month 11)** → Highest sales (~350K)
- **December (Month 12)** → Slight drop but still high (~320K)
- **September (Month 9)** → Strong spike (~300K)

#### Low Months
- **February (Month 2)** → Lowest (~60K)
- **January (Month 1)** → Low start (~94K)
- **April (Month 4)** → Moderate dip (~136K)

---

#### Quarterly View
- **Q4 (Oct–Dec)** →  Peak season (highest overall sales)
- **Q3 (Jul–Sep)** →  Growth phase (rising trend, spike in Sep)
- **Q2 (Apr–Jun)** →  Stable/moderate performance
- **Q1 (Jan–Mar)** →  Weakest quarter

---

### Insight
There is a clear **seasonal pattern**:
- Sales are **low at the start of the year**
- Gradually increase through mid-year
- **Peak strongly in Q4 (likely due to festivals/holiday demand)**

---

### Business Interpretation
- Focus inventory & marketing on **Q3–Q4**
- Investigate demand drop in **Q1**
- Run promotions in early months to smooth revenue

### Customer Segment Analysis

#### Segment with Most Orders (Proxy: Highest Total Sales)
- **Consumer**
  - Highest total sales (~1.14M)
  - Indicates highest volume of orders

#### Segment with Lowest Average Order Value
- **Consumer**
  - Lowest average sales per order (~225)

---

### Insight
The **Consumer segment places the most orders but has the lowest average order value**.  
This suggests a **high-volume, low-value** customer base.

---

### Business Interpretation
- Consumer drives revenue through **scale (volume)**
- Opportunity to:
  - Increase basket size (upselling/cross-selling)
  - Introduce premium offerings to boost AOV

### Loss Analysis

With the available data (Sales only), it is not possible to determine whether products are sold at a loss, as profitability requires cost or profit information.

### Observations
- Some products show **very low total sales**
- Many have **very low order counts (1–4 orders)**

### Insight
These products are **low-performing in terms of revenue and demand**, but cannot be classified as loss-making.

### Recommendation
To identify loss-making products, additional data is required:
- Profit column, or
- Cost (COGS)

Additionally, for consistency:
- Filter products with higher order counts (e.g., >20) before drawing conclusions